jdbc/src/main/java/org/apache/metamodel/jdbc/dialects/DefaultQueryRewriter.java

```diff
@@ -139,13 +143,24 @@ public String rewriteFilterItem(FilterItem item) {
                     }
                 }
 
-                FilterItem replacementFilterItem = new FilterItem(item.getSelectItem(), item.getOperator(), elements);
+                FilterItem replacementFilterItem = new FilterItem(selectItem, item.getOperator(), elements);
                 return super.rewriteFilterItem(replacementFilterItem);
             }
         }
         return super.rewriteFilterItem(item);
     }
-    
+
+    protected String rewriteTimestamp(FilterItem item) {
+        final OperatorType operator = item.getOperator();
+        final SelectItem selectItem = item.getSelectItem();
+        final Object operand = item.getOperand();
+        StringBuilder sb = new StringBuilder();
+        sb.append(selectItem.getSameQueryAlias(false));
+        FilterItem.appendOperator(sb, operand, operator);
+        sb.append("TIMESTAMP \'" + operand.toString() + "\'");
```

Comment:
Did you try with the official JDBC escape syntax? I bet that has a lot better general applicability than this format...